### ISSUE TITLE
Game keeps crashing on linux

### ISSUE THREAD

**Issue by s3agu11** [TeamAOF/All-of-Fabric-6#115]
I hadn't updated since version 1.3 of the pack, but when I updated to 1.4.x it started crashing during startup. It's giving me error code 6.

I am on an arch linux based os, if that helps.

link to the log from my most recent startup: https://mclo.gs/E4onIdx

**Comment by Linguardium** [comment]
this is a glfw error and you will need to compile the latest version of GLFW and point minecraft at it. should be able to grab latest from the arch package repo, then use something like `-Dorg.lwjgl.glfw.libname=/usr/lib/x86_64-linux-gnu/libglfw.so.3.3` to point it

**Comment by s3agu11** [comment]
> this is a glfw error and you will need to compile the latest version of GLFW and point minecraft at it. should be able to grab latest from the arch package repo, then use something like `-Dorg.lwjgl.glfw.libname=/usr/lib/x86_64-linux-gnu/libglfw.so.3.3` to point it

Okay, thanks

**Comment by s3agu11** [comment]
> this is a glfw error and you will need to compile the latest version of GLFW and point minecraft at it. should be able to grab latest from the arch package repo, then use something like `-Dorg.lwjgl.glfw.libname=/usr/lib/x86_64-linux-gnu/libglfw.so.3.3` to point it

So, I have done that, which seems to have resolved that error, but created another one. It says "*** stack smashing detected ***: terminated" and then minecraft crashes. What is the solution for stack smashing?

**Comment by noesterle** [comment]
I believe I am running into a similar Stack Smashing issue while running AOF6 v1.4.2 on Minecraft v1.19.2 with Loader 0.14.14 on Pop!_OS 22.04.

The last few lines of the log are:
```
[18/02/2023 16:50:44 PM] [Render thread/INFO] [Tech Reborn] detected. Loading supported items.
[18/02/2023 16:50:44 PM] [Render thread/INFO] [DragonLoot] detected. Loading supported items.
[18/02/2023 16:50:44 PM] [Render thread/INFO] [Harvest Scythes] has successfully been loaded!
[18/02/2023 16:50:44 PM] [Render thread/INFO] Mixing client.MixinChatComponent from nochatreports.mixins.json into net.minecraft.class_338
[18/02/2023 16:50:44 PM] [Render thread/INFO] Renaming synthetic method lambda$modifyGUIMessage$1()V to md214168$lambda$modifyGUIMessage$1$0 in nochatreports.mixins.json:client.MixinChatComponent from mod nochatreports
[18/02/2023 16:50:44 PM] [Render thread/INFO] Renaming synthetic method lambda$modifyGUIMessage$0(Lnet/minecraft/class_5348;Lnet/minecraft/class_2561;)V to md214168$lambda$modifyGUIMessage$0$1 in nochatreports.mixins.json:client.MixinChatComponent from mod nochatreports
[18/02/2023 16:50:44 PM] [Render thread/INFO] Mixing client.MixinGuiMessageTagIcon from nochatreports.mixins.json into net.minecraft.class_7591$class_7592
[18/02/2023 16:50:44 PM] [Render thread/INFO] Renaming @Invoker method create(Ljava/lang/String;IIIII)Lnet/minecraft/class_7591$class_7592; to create_$md$214168$0 in nochatreports.mixins.json:client.MixinGuiMessageTagIcon from mod nochatreports
[18/02/2023 16:50:44 PM] [Render thread/INFO] Loaded default options for extra-folder
[18/02/2023 16:50:45 PM] [Render thread/INFO] Backend library: LWJGL version 3.3.1 SNAPSHOT
[18/02/2023 16:50:45 PM] [Render thread/INFO] Loaded client.properties
[18/02/2023 16:50:45 PM] *** stack smashing detected ***: terminated
[18/02/2023 16:50:45 PM] Oh no. Minecraft crashed. Please check the logs for any errors and provide these logs when asking for support.
```

The full log can be found at https://www.toptal.com/developers/hastebin

**Comment by Linguardium** [comment]
yes, you also need to update glfw to 3.3.8+ and then set your launcher to use 3.2.2 (important) and point the game at your 3.3.8+ version

**Comment by FadeDave77** [comment]
> yes, you also need to update glfw to 3.3.8+ and then set your launcher to use 3.2.2 (important) and point the game at your 3.3.8+ version

Wow, I don't know how you figured that out, but it sure fixed the problem! Any idea on how to get rid of the annoying "GLFW error 65544: Wayland: Setting window icon not supported." popup window? I don't care about the window icon, but the popup stops the game from loading. Thanks

**Comment by Linguardium** [comment]
you can try removing the packicon from the kubejs config folder though this looks like vanilla would do the same

**Comment by TimeBender25** [comment]
Running into an identical issue, trying to launch AoF6 1.4.3 through Prism Launcher on Mint 20.3 Cinnamon.

I have no idea how to point the game and launcher at a specific glfw version, or even to be sure I've installed glfw 3.3.8 correctly as I don't get that far into Linux often enough to be comfortable knowing I've done something correctly.

I have the `libglfw.so.3.3` file in the right spot, but putting that line up there into the java arguments causes a crash. Should it not go there, or am i doing something else wrong?

Link to the log: https://mclo.gs/iKi6Kuz

**Comment by Linguardium** [comment]
did you also set instance in edit->versions to use LWJGL 3.2.2 using the "change version" option while LWJGL is selected?

**Comment by TimeBender25** [comment]
I hadn't, but setting it only swaps out the crash report for this:

`[20:50:33] [Render thread/INFO]: Loaded default options for extra-folder
[20:50:33] [Render thread/INFO]: Backend library: LWJGL version 3.2.2 build 10
[20:50:33] [Render thread/INFO]: Loaded client.properties
*** buffer overflow detected ***: terminated
Process crashed with exitcode 6.`

**Comment by FadeDave77** [comment]
Did you tick the "use system glfw" setting in settings>minecraft? @TimeBender25

**Comment by TimeBender25** [comment]
> Did you tick the "use system glfw" setting in settings>minecraft? @TimeBender25

I have been, yes. it makes seemingly no difference and I still receive the Buffer Overflow at the same spot of Initialization.
Here's the full log, if it helps at all: https://mclo.gs/5ATrcMO

**Comment by Seegras** [comment]
Same here on Debian Sid. Setting native GLFW alone makes no difference. 

probably, with AoF6 1.4.2 this AoF5 Bug returned: 
https://github.com/TeamAOF/All-of-Fabric-5/issues/507

I was able to start it with 
`-Dorg.lwjgl.glfw.libname=${wherever}/PrismLauncher/lib/libglfw.so.3.4`

${wherever} refers to whatever directory you put PrismLauncher into.

**Comment by Vinesauced** [comment]
Hello, I am having a similar issue, and I am unable to point my game at glfw 3.3.8 using the JVM argument above,
I am using [this glfw package](https://aur.archlinux.org/packages/glfw-wayland-minecraft-libdecoration) on KDE Garuda Linux. And I am using system installation of GLFW.

AOF6 works for a bit when I load it up using the glfw package and no JVM argument, it no longer has the smashing stacks crash when I jump into water, but then it crashes my entire PC when I kill a squid.
The specific JVM argument I am using is `-Dorg.lwjgl.glfw.libname=$/usr/lib/libglfw.so` and the result is a `java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'dynamicfps'!` error. (I have already tried using libglfw.so.3.4)
this is the crash report if anyone wants to see the full thing, [https://mclo.gs/Oh44deC](https://mclo.gs/Oh44deC). Is this crash something completely unrelated to glfw? if so why doesn't it happen without that JVM argument?

**Comment by Seegras** [comment]
> The specific JVM argument I am using is `-Dorg.lwjgl.glfw.libname=$/usr/lib/libglfw.so.3.4` 

That does not make sense. $-something is used to denote a variable. You're supposed to put in the variable name after the $-sign. So either this $ is superfluous, because you're using the system path (starting with /usr/lib ... ), or you're missing the rest of the path there.

**Comment by Vinesauced** [comment]
Syntax really matters a lot here.  Removing the $ fixed it that error, thank you.
Setting the JVM argument didn't help anything, my GPU still dies when I kill a squid and other random events (I have killed several squids in this world before and they never gave me this crash.) There is no indication of anything going wrong in the MC log, but there is this message in dmesg
`[  888.982724] [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx_0.0.0 timeout, signaled seq=123091, emittedseq=123093
[  888.982924] [drm:amdgpu_job_timedout [amdgpu]] *ERROR* Process information: process java pid 38757 thread java:cs0 pid 39722
[  888.983070] amdgpu 0000:03:00.0: amdgpu: GPU reset begin!`
Doing a bit of research, this error could be linked with a hardware issue with my 6800 XT, but I am very doubtful of that as I had not gotten this error before, even when I played the more resource intensive All The Mods 8.

**Comment by Arr4y** [comment]
I made a similar post of this for AOF5. I am running Garuda Linux (Arch). But I crashed instantly when loading AOF5 (this happened in AOF6 when I had it too). Error code 6 popped up as well.

When I looked at the crash dump, this line stood out to me.
`Problematic frame: C  [libglfw.so+0x1674b]`

I am using Prism Launcher. I enabled "Use system installation of GLFW" in the settings for my AOF5 instance. And the game stopped crashing completely. Perhaps this might work for AOF6 as well?

**Comment by Funmud1** [comment]
Hello I am also having the same issue on Ubuntu. I am using the native Minecraft launcher and the latest version of AOF6.
The above solutions did not work for me, either I did them incorrectly or just did not work.
I believe this is related to the issue:
# Problematic frame:
# C  [libglfw.so+0x2257b]
